### PR TITLE
XRDDEV-984 

### PR DESCRIPTION
**Pull request by bertofl90**
Add volume support for xroad config
https://jira.niis.org/browse/XRDDEV-984?filter=-1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,3 +40,4 @@ The script setup_security_server_sidecar.sh will:
 - Update X-Road configuration on startup if the installed version on image has been updated.
 - Generate serverconf database and properties file with default username (serverconf) and random password.
 - Generate internal and admin UI TLS keys and certificates.
+- Add volume support for etc/xroad config
```
Comment (raulmartinez-leadin) [inline README.md:43]: Maybe reword this sentence so that it makes sense with the rest of the steps that the script does, something like
```suggestion
- Create sidecar-config directory on the host and mount it into the /etc/xroad config directory on the container.
```

File: setup_security_server_sidecar.sh [hunk 2]
```diff
@@ -24,11 +24,10 @@ httpport=$(($2 + 1))
 
 # Create xroad-network to provide container-to-container communication
 docker network inspect xroad-network >/dev/null 2>&1 || docker network create -d bridge xroad-network
-
 echo "=====> Build sidecar image"
-docker build -f sidecar/Dockerfile -t xroad-sidecar-security-server-image sidecar/
+docker build -f sidecar/Dockerfile -t xroad-sidecar-security-server-image sidecar/ 
 echo "=====> Run container"
-docker run --detach -p $2:4000 -p $httpport:80 -p 5588:5588 --network xroad-network -e XROAD_TOKEN_PIN=$3 -e XROAD_ADMIN_USER=$4 -e XROAD_ADMIN_PASSWORD=$5 --name $1 xroad-sidecar-security-server-image
+docker run -v sidecar-config:/etc/xroad --detach -p $2:4000 -p $httpport:80 -p 5588:5588 --network xroad-network -e XROAD_TOKEN_PIN=$3 -e XROAD_ADMIN_USER=$4 -e XROAD_ADMIN_PASSWORD=$5 --name $1 xroad-sidecar-security-server-image
```
Comment (jhyoty) [inline setup_security_server_sidecar.sh:30]: This moves the config files into a volume, but the serverconf database is still stored in the container (/var/lib/postgresql/10/main) and will be lost if the image is updated.

Comment (bertofl90) [reply setup_security_server_sidecar.sh]: I've added a new volume for this folder

File: README.md [hunk 3]
```diff
@@ -44,7 +43,7 @@ Minimum recommended docker engine configuration to run the security server sidec
  1.5    | TCP 5500                                  | Ports for inbound connections (from the external network to the security server)<br> Message exchange between security servers
  &nbsp; | TCP 5577                                  | Ports for inbound connections (from the external network to the security server)<br> Querying of OCSP responses between security servers
  1.6    | TCP 5500                                  | Ports for outbound connections (from the security server to the external network)<br> Message exchange between security servers
- &nbsp; | TCP 5577                                  | Ports for outbound connections (from the security server to the external network)<br> Querying of OCSP responses between security servers
+ &nbsp; | TCP 5577                                  | Ports for outbound connections (from the security server to the exte>>>>>>> masterrnal network)<br> Querying of OCSP responses between security servers
```
Comment (raulmartinez-leadin) [inline README.md:46]: Typo here, seems like a remain of the merge conflict. Please correct.

File: README.md [hunk 4]
```diff
@@ -87,15 +87,15 @@ The script setup_security_server_sidecar.sh will:
...
+- Create sidecar-config directory on the host and mount it into the /etc/xroad config directory on the container.
 ## 2 Security Server Sidecar Initial Configuration
 
 ### 2.1 Reference Data
 
  **Ref** | **Value**                                                | **Explanation**
  ---- | ----------------------------------------------------------- | -------------------------------------------------------
  2.1  | &lt;global configuration anchor file&gt; or &lt;URL&gt;     | Global configuration anchor file or provider URL (1) (2)
- 2.2  | &lt;security server owner's member class&gt;<br>E.g.<br> COM - Commercial<br> ORG - Organisation            | Member class of the security server owner for the sidecar (2)
+ 2.2  | &lt;security server owner's member class&gt;<br>E.g.<br> COM- Create sidecar-config directory on the host and mount it into the /etc/xroad config directory on the container. - Commercial<br> ORG - Organisation            | Member class of the security server owner for the sidecar (2)
```
Comment (raulmartinez-leadin) [inline README.md:98]: Line slipped here, I don't think it's supposed to be there.

File: README.md [hunk 5]
```diff
@@ -25,10 +25,9 @@ The Security Server sidecar installation requires an existing installation of Do
 
 Minimum recommended docker engine configuration to run the security server sidecar container:
 
-- CPUs: 2
-- Memory: 2 GiB
-- Swap: 1 GiB
-- Disk space: 2 GiB
+- CPUs: 4
+- Memory: 8 G
```
Comment (raulmartinez-leadin) [inline README.md:29]: Seems the minimum required specifications were updated here? If so, make sure to keep Disk space requirement and change G -> GiB

File: README.md [hunk 6]
```diff
@@ -138,3 +138,4 @@ Then, if the configuration is successfully downloaded, the system asks for the f
 - The security server configuration files (located in `/etc/xroad/`) are currently stored inside the sidecar container. Functionality to provide volume support will be added in future releases.
```
Comment (raulmartinez-leadin) [inline README.md:138]: This line about the not supported functionality about storing configuration in a volume is not needed anymore.

File: README.md [hunk 7]
```diff
@@ -124,6 +125,18 @@ The following configuration is needed on the remote database server to allow ext
...
+It is possible to configure security server sidecar to use volume support. This will allow us to  create sidecar-config and sidecar-config-db directory on the host and mount it into the /etc/xroad and /var/lib/postgresql/10/main  config  directories on the container.
+For adding volume support we have to modify the docdker run sentence inside the setup_security_server_sidecar.sh script and add the volume support:
+**-v (sidecar-config-volume-name):/etc/xroad -v (sidecar-config-db-volume-name):/var/lib/postgresql/10/main **
+For example
+  ```bash
+  [...]
+    docker run  -v sidecar-config:/etc/xroad -v sidecar-config-db:/var/lib/postgresql/10/main -detach -p $2:4000 -p $httpport:80 -p 5588:5588 --network xroad-network -e XROAD_TOKEN_PIN=$3 -e XROAD_ADMIN_USER=$4 -e XROAD_ADMIN_PASSWORD=$5 -e XROAD_DB_HOST=$postgresqlhost -e XROAD_DB_PORT=$postgresqlport -e XROAD_DB_PWD=$XROAD_DB_PASSWORD --name $1 xroad-sidecar-security-server-image
+  [...]
+  ```
+
```
Comment (jhyoty) [inline README.md:139]: There are some typos here (at least Volumme, suport, docdker). Please also verify that the document is rendered correctly when viewed on github.
